rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/history/HistoryStore.scala

```diff
@@ -1,12 +1,45 @@
 package coop.rchain.rspace.nextgenrspace.history
 
+import cats.implicits._
+import cats.effect.Sync
 import coop.rchain.rspace.Blake2b256Hash
 import coop.rchain.rspace.nextgenrspace.history.History._
+import coop.rchain.shared.AttemptOps.RichAttempt
 
 trait HistoryStore[F[_]] {
   def put(tries: List[Trie]): F[Unit]
 
   def get(key: Blake2b256Hash): F[Trie]
+
+  def close(): F[Unit]
 }
 
-object HistoryStore {}
+object HistoryStoreInstances {
+  def historyStore[F[_]: Sync](store: Store[F]) = new HistoryStore[F] {
+    // TODO put list
+    override def put(tries: List[Trie]): F[Unit] =
+      tries
+        .map { t =>
+          val key = Trie.hash(t)(History.codecTrie)
+          (key, t)
+        }
+        .traverse {
+          case (key, value) =>
+            val bytes = History.codecTrie.encode(value).get
+            store.put(key, bytes)
+        }
+        .as(())
```

Comment:
this can be simplified:
```scala
      tries
        .traverse {
          value =>
            val key = Trie.hash(value)(History.codecTrie)
            val bytes = History.codecTrie.encode(value).get
            store.put(key, bytes)
        }
        .as(())
```